common/app/model/Cached.scala

```diff
@@ -36,7 +36,7 @@ object CacheTime {
 
 object Cached extends implicits.Dates {
 
-  private val cacheableStatusCodes = Seq(200, 404)
+  private val cacheableStatusCodes = Seq(200, 404, 304)
```

Comment:
I'm not really sure about this, we are saying 304 is a cacheable status code in our world, but that bascially means we'll add cache headers when we call Cached.apply.  That calls the cacheHeaders method which will add headers including an ETag.  however since we have deemed this a WithoutRevalidationResult above, this means it will have a random etag generated I think.
I think doing it this way is a bit of a trick, as the result we are returning isn't really a Revalidatable or Without revalidation result, it's a don't-know-just-panic-theres-no-result.  It might be more appropriate to make a Third kind of result other than RevalidatableResult and WithoutRevalidationResult.  That would be a PanicReuseExistingResult or something, and would trigger the Cached function to do the right thing with the headers.  Then the panic filter just worries about whether it should panic, rather than what panicing actually does.  The cache function can then be the source of decision on what headers do what in terms of caching.

Reply:
...one month later, I've added something along these lines, thanks John!